File: frontend/src/app/core/use-case/get-accident-points.use-case.ts
import { Injectable, inject } from '@angular/core';
import { GetAccidentPointsPort } from '../port/out/accident.port';
import type { AccidentPoint } from '../domain/accident.model';
import { Observable } from 'rxjs';

/**
 * Caso de uso: obtener puntos de accidentes por provincias.
 * Depende del puerto de salida (adaptador HTTP en runtime).
 */
@Injectable({ providedIn: 'root' })
export class GetAccidentPointsUseCase {
  private readonly port = inject(GetAccidentPointsPort);

  getByProvinces(provinces: string[]): Observable<AccidentPoint[]> {
    return this.port.getByProvinces(provinces);
  }
}

File: backend/src/main/java/com/joaquin/proyecto_backend/application/port/in/GetAccidentPointsUseCase.java
package com.joaquin.proyecto_backend.application.port.in;

import com.joaquin.proyecto_backend.domain.AccidentPoint;

import java.util.List;

/**
 * Puerto de entrada (caso de uso): listar puntos de accidentes por provincias.
 */
public interface GetAccidentPointsUseCase {

    List<AccidentPoint> getByProvinces(List<String> provinces);
}
